### PR TITLE
Simplified various assignment statements.

### PR DESCRIPTION
**Pull request by ngnpope**
Tidied up the following:

- Used more augmented assignment statements.
- Avoided assignment followed by immediate return.

Identified using the following commands:

```console
$ git grep '\(\<[_a-zA-Z0-9]\+\>\) *= *\1 *[-+/*^%&|<>@]' ':*.py'
$ pcre2grep --line-number --multiline --recursive \
        "(?s)(\n +)(\w+) = [^\n]+\1return \2;?$" \
        django docs extras js_tests scripts tests
```

### CODE REVIEW

File: django/contrib/admin/static/admin/js/admin/RelatedObjectLookups.js [hunk 1]
```diff
@@ -26,13 +26,11 @@
...
     function addPopupIndex(name) {
-        name = name + "__" + (popupIndex + 1);
-        return name;
+        return name + "__" + (popupIndex + 1);
     }
 
     function removePopupIndex(name) {
-        name = name.replace(new RegExp("__" + (popupIndex + 1) + "$"), '');
-        return name;
+        return name.replace(new RegExp("__" + (popupIndex + 1) + "$"), '');
```
[the remark this answers is not in the document]
Comment (ngnpope) [reply django/contrib/admin/static/admin/js/admin/RelatedObjectLookups.js:33]: I've updated this to create a separate commit.

I've not separated the other piece mentioned - I think it's okay to leave.

**Comment by ngnpope** [comment]
> I think the term you're looking for is [augmented assignment statement](https://docs.python.org/3/reference/simple_stmts.html#augmented-assignment-statements) grin

Yeah... I was a little bit handwavy in my description.
That's the bit I always struggle to find in the Python docs!

Thanks @shangxiao 👍🏻

**Review by shangxiao — APPROVED** [review]
lgtm 🎉

File: docs/topics/http/middleware.txt [hunk 2]
```diff
@@ -340,14 +340,12 @@ Here's an example of how to create a middleware function that supports both::
...
-                response = await get_response(request)
-                return response
+                return await get_response(request)
 
         else:
             def middleware(request):
                 # Do something here!
-                response = get_response(request)
-                return response
+                return get_response(request)
```
Comment (felixxm) [inline docs/topics/http/middleware.txt:348]: I would revert this change. As far as I'm aware there is an intention to do something with the response in between `get_response()` and `return`, so separate statements make more sense.

Comment (ngnpope) [reply docs/topics/http/middleware.txt:348]: Reverted.

Comment (pauloxnet) [reply docs/topics/http/middleware.txt]: > I would revert this change. As far as I'm aware there is an intention to do something with the response in between `get_response()` and `return`, so separate statements make more sense.

If this is the case we can move `# Do something here!` between `get_response()` and `return`

Comment (ngnpope) [reply docs/topics/http/middleware.txt]: I thought about that, but realised that really you'd be wanting something like:
```
            def middleware(request):
                # Do something before processing the view...
                response = get_response(request)
                # Do something after processing the view...
                return response
```
And then I decided that it felt beyond the scope of what I was trying to do. If we were to do that I'd make it a separate commit. Let me know if you'd like this change, @felixxm.

Comment (felixxm) [reply docs/topics/http/middleware.txt]: > And then I decided that it felt beyond the scope of what I was trying to do. If we were to do that I'd make it a separate commit. Let me know if you'd like this change, @felixxm.

The current wording works for me, I'd leave it as it is :shrug:

**Comment by felixxm** [comment]
@ngnpope Thanks :+1:
